atlasdb-ete-tests/build.gradle

```diff
@@ -101,43 +101,56 @@ task prepareForEteTests(type: Copy, dependsOn: 'distTar') {
     rename { filename -> 'atlasdb-ete-snapshot.tgz' }
 }
 
-task longTest(type: Test) {
-    dependsOn prepareForEteTests
-    include '**/MultiCassandraTestSuite.class'
-}
-
-task timeLockTest(type: Test) {
+// standalone test with its own task
+// is very expensive and so separate task is used to control when/where it is executed
+task timeLockMigrationTest(type: Test) {
     dependsOn prepareForEteTests, ':timelock-server-distribution:dockerTag'
-    include '**/*TimeLock*.class'
-    exclude '**/*TimeLockMigration*.class'
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.standalone.TimeLockMigrationEteTest'
+    }
 }
 
-task timeLockMigrationTest(type: Test) {
-    dependsOn prepareForEteTests, ':timelock-server-distribution:dockerTag'
-    include '**/*TimeLockMigration*.class'
+// test suites with their own test tasks
+// not sure why some have it and others do not
+
+task singleClientWithEmbeddedAndThreeNodeCassandraTest(type: Test) {
+    dependsOn prepareForEteTests
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndThreeNodeCassandraTestSuite'
+    }
 }
 
-task dbkvsTest(type: Test) {
+task singleClientWithEmbeddedAndPostgresTest(type: Test) {
     dependsOn prepareForEteTests
-    include '**/DbKvsTestSuite.class'
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndPostgresTestSuite'
+    }
 }
 
-task oracleTest(type: Test) {
+task singleClientWithEmbeddedAndOracleTest(type: Test) {
     dependsOn prepareForEteTests
-    include '**/OracleDbKvsEteTestSuite.class'
+    filter {
+        includeTestsMatching 'com.palantir.atlasdb.ete.suites.SingleClientWithEmbeddedAndOracleTestSuite'
+    }
 }
 
 test {
-    dependsOn longTest, prepareForEteTests, timeLockTest, dbkvsTest
-    exclude '**/MultiCassandraTestSuite.class'
-
-    exclude '**/*TimeLock*.class'
-    // Technically covered by previous line, but explicitly avoiding in case of refactor
-    exclude '**/*TimeLockMigration*.class'
-
-    exclude '**/*EteTest.class'
-    exclude '**/DbKvsTestSuite.class'
-    exclude '**/OracleDbKvsEteTestSuite.class'
+    // this list explicitly doesn't include timelockMigration because it's too expensive to run on every PR
+    // this list explicitly doesn't include singleClientWithEmbeddedAndOracleTest; not sure why
+    // :timelock-server-distribution:dockerTag needed by any ETE test that spins up a timelock-server
+    dependsOn prepareForEteTests, ':timelock-server-distribution:dockerTag', singleClientWithEmbeddedAndThreeNodeCassandraTest, singleClientWithEmbeddedAndPostgresTest
```

Comment:
yea I don't really understand the back-story here. but I can clean it up if people agree.

Reply:
One benefit of this setup is that `atlasdb-ete-tests:test` task will run all tests by default. We should either:
- Add the other standalone test tasks as dependencies here, and rely on the Circle command task exclusions to avoid empty tests.
- Remove all standalone test task dependencies and rely on Circle commands to run the standalone tasks.

The former feels like the safer approach.

We probably do want these test to run separately - timings are below:
- `singleClientWithEmbeddedAndThreeNodeCassandraTest` takes 2 minutes, 32 seconds ([scan](https://scans.gradle.com/s/cipmxkayaiy6i/tests/overview#:atlasdb-ete-tests:singleclientwithembeddedandthreenodecassandratest-1))
- `singleClientWithEmbeddedAndPostgresTest` takes 1 minute, 3 seconds ([scan](https://scans.gradle.com/s/nqmdgecgzkld4/tests/overview#:atlasdb-ete-tests:singleclientwithembeddedandpostgrestest-2))
- `singleClientWithEmbeddedAndOracleTest` takes 3 minutes, 28 seconds ([scan](https://scans.gradle.com/s/uwnzpkfgzpu4e/tests/overview#:atlasdb-ete-tests:singleclientwithembeddedandoracletest-0))
- All other `atlasdb-ete-tests` tests take 2 minutes, 28 seconds ([scan](https://scans.gradle.com/s/5tbtp7elvguuk/tests/overview#:atlasdb-ete-tests:test-0))

Reply:
We might want to bundle the one-off tests behind a separate task, that way we simplify the configuration by having a single task as:
- A dependency of the `test` task
- An exclusion in the Circle config